image/pict.cpp

```diff
@@ -166,11 +166,26 @@ void PICTDecoder::o_longText(Common::SeekableReadStream &stream) {
 
 void PICTDecoder::o_bitsRgn(Common::SeekableReadStream &stream) {
 	// Copy unpacked data with clipped region (8bpp or lower)
-	unpackBitsRgn(stream, false);
+
+	PixMap pixMap = readRowBytes(stream, false);
+	bool isPixMap = (pixMap.rowBytes & 0x8000) ? true : false;
+	pixMap.rowBytes = pixMap.rowBytes & 0x7fff;
+
+	if (isPixMap)
+		unpackBitsRect(stream, true, pixMap);
+	else
+		unpackBitsRgn(stream, false);
```

Comment:
Since this whole block is repeated over and over, it would be good to move all this code into a separate function, e.g. `unpackBitsRectOrRgn()`

Reply:
ok